apps/studio/components/interfaces/Storage/FilesBuckets.tsx

```diff
@@ -0,0 +1,172 @@
+import { Edit, FolderOpen, MoreVertical, Search, Trash2 } from 'lucide-react'
+
+import { useParams } from 'common'
+import { Bucket, useBucketsQuery } from 'data/storage/buckets-query'
+import { useState } from 'react'
+import {
+  Button,
+  Card,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from 'ui'
+import { Input } from 'ui-patterns/DataInputs/Input'
+import { GenericSkeletonLoader } from 'components/ui/ShimmeringLoader'
+import { CreateBucketModal } from './CreateBucketModal'
+import { DeleteBucketModal } from './DeleteBucketModal'
+import { EditBucketModal } from './EditBucketModal'
+import { EmptyBucketModal } from './EmptyBucketModal'
+import { EmptyBucketState } from './EmptyBucketState'
+
+export const FilesBuckets = () => {
+  const { ref } = useParams()
+
+  const [modal, setModal] = useState<'edit' | 'empty' | 'delete' | null>(null)
+  const [selectedBucket, setSelectedBucket] = useState<Bucket>()
+  const [filterString, setFilterString] = useState('')
+
+  const { data: buckets = [], isLoading } = useBucketsQuery({ projectRef: ref })
+  const filesBuckets = buckets
+    .filter((bucket) => !('type' in bucket) || bucket.type === 'STANDARD')
+    .filter((bucket) =>
+      filterString.length === 0
+        ? true
+        : bucket.name.toLowerCase().includes(filterString.toLowerCase())
+    )
+  return (
+    <>
+      {!isLoading &&
+      buckets.filter((bucket) => !('type' in bucket) || bucket.type === 'STANDARD').length === 0 ? (
+        <EmptyBucketState bucketType="files" />
+      ) : (
+        <div className="pt-12 flex flex-col gap-y-4">
```

Comment:
`pt-12` is visually to large of a gap but is necessary here because any smaller would be misaligned with other tabs.

Something to polish in post.